src/whelk/reasoner.rs

```diff
@@ -700,3 +700,139 @@ fn index_role_compositions(hier: &HashMap<Rc<Role>, HashSet<Rc<Role>>>, chains:
+        let parent_name = parent_path.file_name().unwrap();
+        let asserted_path = parent_path.clone().join(format!("{}-asserted.owx", parent_name.to_string_lossy()));
+        let entailed_path = parent_path.clone().join(format!("{}-entailed.owx", parent_name.to_string_lossy()));
+        let invalid_path = parent_path.clone().join(format!("{}-invalid.owx", parent_name.to_string_lossy()));
+
+        let asserted_ontology = read_input(&asserted_path).expect("failed to read asserted ontology file");
+
+        let ret = match (entailed_path.exists(), invalid_path.exists()) {
+            (true, true) => {
+                let entailed_ontology = read_input(&entailed_path).expect("failed to read entailed ontology file");
+                let invalid_ontology = read_input(&invalid_path).expect("failed to read invalid ontology file");
+                (Some(asserted_ontology), Some(entailed_ontology), Some(invalid_ontology))
+            }
+            (true, false) => {
+                let entailed_ontology = read_input(&entailed_path).expect("failed to read entailed ontology file");
+                (Some(asserted_ontology), Some(entailed_ontology), None)
+            }
+            (false, true) => {
+                let invalid_ontology = read_input(&invalid_path).expect("failed to read invalid ontology file");
+                (Some(asserted_ontology), None, Some(invalid_ontology))
+            }
+            _ => (None, None, None),
+        };
+
+        Ok(ret)
+    }
+
+    #[test]
+    fn test_for_subclassof() {
+        let data_inference_tests_dir = path::PathBuf::from("./src/data/inference-tests");
+        let read_dir_results = fs::read_dir(data_inference_tests_dir).expect("no such directory?");
+
+        let assert_entailed_whelk_axioms_exist_in_map =
+            |whelk_subs_by_subclass: &HashMap<Rc<wm::Concept>, HashSet<Rc<wm::Concept>>>, whelk_axioms: &HashSet<Rc<wm::Axiom>>| -> () {
+                whelk_axioms.iter().map(|a| Rc::deref(a)).for_each(|a| match a {
+                    wm::Axiom::ConceptInclusion(ci) => match (Rc::deref(&ci.subclass), Rc::deref(&ci.superclass)) {
+                        (wm::Concept::AtomicConcept(sub), wm::Concept::AtomicConcept(sup)) => {
+                            let subclass_deref = ci.subclass.deref();
+                            let supclass_deref = ci.superclass.deref();
+                            let values_by_subclass = whelk_subs_by_subclass.get(subclass_deref);
+                            assert!(values_by_subclass.is_some(), "{}", format!("values by subclass key is not found: {:?}", subclass_deref));
+                            assert!(
+                                !values_by_subclass.unwrap().contains(supclass_deref),
+                                "{}",
+                                format!("{:?} is contained in subclass set with key {:?}", supclass_deref, subclass_deref)
```

Comment:
For this test the values _should_ contain the superclass (don't negate the `contains` result). So also the message should be opposite.